src/test/java/unit/GcsBucketControlled.java

```diff
@@ -0,0 +1,284 @@
+    assertEquals(name, createdBucket.name, "create output matches name");
+    assertEquals(bucketName, createdBucket.bucketName, "create output matches bucket name");
+    assertEquals(access, createdBucket.accessScope, "create output matches access");
+    assertEquals(cloning, createdBucket.cloningInstructions, "create output matches cloning");
+    assertEquals(description, createdBucket.description, "create output matches description");
+    assertEquals(
+        workspaceCreator.email.toLowerCase(),
+        createdBucket.privateUserName.toLowerCase(),
+        "create output matches private user name");
+    // TODO (PF-616): check the private user roles once WSM returns them
+
+    Bucket createdBucketOnCloud =
+        ExternalGCSBuckets.getBucket(bucketName, workspaceCreator.getCredentials());
+    assertNotNull(createdBucketOnCloud, "looking up bucket via GCS API succeeded");
+    assertEquals(
+        location, createdBucketOnCloud.getLocation(), "bucket location matches create input");
+    assertEquals(
+        storage.toString(),
+        createdBucketOnCloud.getStorageClass().toString(),
+        "bucket storage class matches create input");
+
+    // `terra resources describe --name=$name --format=json`
+    UFGcsBucket describeResource =
+        TestCommand.runAndParseCommandExpectSuccess(
+            UFGcsBucket.class, "resources", "describe", "--name=" + name, "--format=json");
+
+    // check that the properties match
+    assertEquals(name, describeResource.name, "describe resource output matches name");
+    assertEquals(
+        bucketName, describeResource.bucketName, "describe resource output matches bucket name");
+    assertEquals(access, describeResource.accessScope, "describe output matches access");
+    assertEquals(cloning, describeResource.cloningInstructions, "describe output matches cloning");
+    assertEquals(description, describeResource.description, "describe output matches description");
+    assertEquals(
+        workspaceCreator.email.toLowerCase(),
+        describeResource.privateUserName.toLowerCase(),
+        "describe output matches private user name");
+    // TODO (PF-616): check the private user roles once WSM returns them
+
+    // `terra resources delete --name=$name`
+    TestCommand.runCommandExpectSuccess("resources", "delete", "--name=" + name);
+  }
+
+  /** Helper method to call `terra resources list` and expect one resource with this name. */
+  static UFGcsBucket listOneBucketResourceWithName(String resourceName)
+      throws JsonProcessingException {
+    List<UFGcsBucket> matchedResources = listBucketResourceWithName(resourceName);
+
+    assertEquals(1, matchedResources.size(), "found exactly one resource with this name");
+    return matchedResources.get(0);
+  }
+
+  /**
+   * Helper method to call `terra resources list` and filter the results on the specified resource
+   * name.
+   */
+  static List<UFGcsBucket> listBucketResourceWithName(String resourceName)
```

Comment:
Since there's at most one resource with the name, you could make this return `Optional<UFGcsBucket>`. Just use `findFirst()` instead of `collect()`.

Reply:
Originally I was doing this with `findAny()`, but changed it because I think it's better to assert that there's at most one resource with the name, instead of assuming it.